pkg/compose/wait.go

```diff
@@ -0,0 +1,64 @@
+/*
+   Copyright 2020 Docker Compose CLI authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package compose
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/compose-spec/compose-go/types"
+	"github.com/docker/compose/v2/pkg/api"
+)
+
+func (s *composeService) Wait(ctx context.Context, project *types.Project, options api.WaitOptions) (int64, error) {
+	containers, err := s.getContainers(ctx, project.Name, oneOffInclude, false, options.Services...)
+	if err != nil {
+		return 0, err
+	}
+
+	var returnErrorChan = make(chan error)
+	var statusCode int64
+	for _, c := range containers {
+		c := c
+		go func() {
```

Comment:
could use an errorGroup (like we do in other places in compose) to capture first error and stop other API calls by cancelable context